packages/elasticsearch-plugin/src/elasticsearch.service.ts

```diff
@@ -293,18 +335,37 @@ export class ElasticsearchService implements OnModuleInit, OnModuleDestroy {
         );
         elasticSearchBody.from = 0;
         elasticSearchBody.size = 0;
-        elasticSearchBody.aggs = {
-            collection: {
-                terms: {
-                    field: 'collectionIds',
-                    size: this.options.searchConfig.collectionMaxSize,
+        if (groupByProduct) {
+            elasticSearchBody.aggs = {
+                collection: {
+                    terms: {
+                        field,
+                        size: this.options.searchConfig.collectionMaxSize,
+                    },
+                    aggs: {
+                        total: {
+                            cardinality: {
+                                field: `productId`,
+                            },
+                        },
+                    },
                 },
-            },
-        };
+            };
+        } else {
+            elasticSearchBody.aggs = {
```

Comment:
The `elasticSearchBody.aggs.collection.terms` part of these 2 `if` branches is identical right? In that case, it can be extracted outside the `if` statement:
```ts
elasticSearchBody.aggs = {
    collection: {
        terms: {
            field,
            size: this.options.searchConfig.collectionMaxSize,
        },
    },
};
if (groupByProduct) {
    elasticSearchBody.aggs.collection.aggs = { /*.... */ };
}
```